pkg/github/github.go

```diff
@@ -596,3 +597,40 @@ func (g *GitHubClient) withRetries(ctx context.Context, retryFunc retry.RetryFun
 	}
 	return nil
 }
+
+type TokenSourceInputs struct {
+	GitHubToken string
+
+	GitHubAppID             string
+	GitHubAppPrivateKeyPEM  string
+	GitHubAppInstallationID string
+	GitHubRepo              string
+	Permissions             map[string]string
+}
+
+// TokenSource creates a token source from a GitHub token or GitHub App used for
+// authenticating a github client.
+func TokenSource(ctx context.Context, inputs *TokenSourceInputs) (githubauth.TokenSource, error) {
+	if inputs.GitHubToken != "" {
+		githubTokenSource, err := githubauth.NewStaticTokenSource(inputs.GitHubToken)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create github static token source: %w", err)
+		}
+		return githubTokenSource, nil
+	} else {
```

Comment:
Its idiomatic to not have the else here, since you return above in the if, less indentation.